googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1337,17 +1346,24 @@ constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
 
 #define GTEST_TEST_NO_THROW_(statement, fail) \
   GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
-  if (::testing::internal::AlwaysTrue()) { \
+  if (::testing::internal::TrueWithString gtest_msg = std::string()) { \
     try { \
       GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement); \
     } \
+    catch (std::exception const& e) { \
+      gtest_msg.value = " std::exception-derived exception with description: "; \
+      gtest_msg.value += e.what(); \
+      gtest_msg.value += "."; \
+      goto GTEST_CONCAT_TOKEN_(gtest_label_testnothrow_, __LINE__); \
+    } \
     catch (...) { \
+      gtest_msg.value = "."; \
```

Comment:
What is the purpose of this?

Reply:
To match previous behavior. See unit-tests

Reply:
updated to be more explicit